tools/tpm2_changepps.c

```diff
@@ -0,0 +1,68 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include "log.h"
+#include "tpm2.h"
+#include "tpm2_options.h"
+#include "tpm2_tool.h"
+
+typedef struct dictionarylockout_ctx dictionarylockout_ctx;
+struct dictionarylockout_ctx {
+    struct {
+        const char *ctx_path;
+        const char *auth_str;
+        tpm2_loaded_object object;
+    } auth_hierarchy;
+};
+
+static dictionarylockout_ctx ctx = {
+    /*
+     * Only platform authorization is needed
+     */
+    .auth_hierarchy.ctx_path = "p",
+};
+
+
+bool on_arg(int argc, char **argv) {
+
+    if (argc > 1) {
+        LOG_ERR("Specify a single auth value");
+        return false;
+    }
+
+    if (!argc) {
+        //empty auth
+        return true;
+    }
+
+    ctx.auth_hierarchy.auth_str = argv[0];
+
+    return true;
+}
+
+bool tpm2_tool_onstart(tpm2_options **opts) {
+
+    *opts = tpm2_options_new(NULL, 0, NULL, NULL, on_arg, 0);
+
+    return *opts != NULL;
+}
+
+tool_rc tpm2_tool_onrun(ESYS_CONTEXT *ectx, tpm2_option_flags flags) {
+
+    UNUSED(flags);
+
+    tool_rc rc = tpm2_util_object_load_auth(ectx, ctx.auth_hierarchy.ctx_path,
+            ctx.auth_hierarchy.auth_str, &ctx.auth_hierarchy.object, false,
+            TPM2_HANDLE_FLAGS_P);
+    if (rc != tool_rc_success) {
+        LOG_ERR("Invalid platform hierarchy authorization");
+        return rc;
+    }
+
+    return tpm2_changepps(ectx, &ctx.auth_hierarchy.object);
```

Comment:
make this just take a tpm2_session * and hardcode the command to the proper ESYS_TR handle